src/Microsoft.ApplicationInsights/Extensibility/Implementation/CorrelationLookup/DictionaryCorrelationIdProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.CorrelationLookup
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Correlation Provider that holds a dictionary of ikeys to correlationIds.
+    /// </summary>
+    public class DictionaryCorrelationIdProvider : ICorrelationIdProvider
```

Comment:
this is cool! It enables config-based definition of correlation ID mapping. Love it